src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -329,4 +310,36 @@ private Map<String, String> extractFromChatParameters(Map<String, String> parame
         }
         return parameters;
     }
+
+    private String parseOutput(String llmOutput, String indexName) {
+        String ppl;
+        Pattern pattern = Pattern.compile("<ppl>((.|[\\r\\n])+?)</ppl>");
+        Matcher matcher = pattern.matcher(llmOutput);
+
+        if (matcher.find()) {
+            ppl = matcher.group(1).replaceAll("[\\r\\n]", " ").replaceAll("ISNOTNULL", "isnotnull").trim();
+        } else { // logic for only ppl returned
+            int sourceIndex = llmOutput.indexOf("source=");
+            if (sourceIndex != -1) {
+                llmOutput = llmOutput.substring(sourceIndex);
+
+                // Splitting the string at "|"
+                String[] lists = llmOutput.split("\\|");
+
+                // Modifying the first element
+                if (lists.length > 0) {
+                    lists[0] = "source=" + indexName;
+                }
+
+                // Joining the string back together
+                ppl = String.join("|", lists);
+            } else {
+                ppl = llmOutput;
```

Comment:
Is the ppl still correct when it's missing `source=xxx`?

Reply:
No. Should we raise error here? In the logic, it will raise error when executing if we don't start with source=xxx.

Reply:
`Executing` means invoking the ppl transport action, right? If so, I suggest to raise error here since it's not going to success and we can avoid network communication.